libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/BaseRoadObject.kt

```diff
@@ -0,0 +1,69 @@
+package com.mapbox.navigation.core.trip.model.roadobject
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectLocation
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectType
+
+/**
+ * Abstract class that serves as a base for all road objects.
+ *
+ * @param id id of the object
+ * @param objectType constant describing the object type, see [RoadObjectType].
+ * @param length length of the object, null if the object is point-like.
+ * @param location location of the road object.
+ * @param provider provider of the road object
+ */
+abstract class BaseRoadObject(
```

Comment:
Why do we need this class if we already have an interface that does exactly the same thing? This class doesn't seem to provide any additional logic over the interface definition, we probably only need one or the other.

Reply:
I think I understand why this is needed, but maybe the class can be `internal` then?

Reply:
it is stored in `core` and it has 
` internal val nativeRoadObject: com.mapbox.navigator.RoadObject`
so, I can access this val from `core` module and pass to the native.

Reply:
I made the constructor internal.